src/jdk.compiler/share/classes/com/sun/tools/javac/tree/TreeInfo.java

```diff
@@ -276,6 +276,27 @@ public static boolean isDiamond(JCTree tree) {
         }
     }
 
+    /** Return true if a default expression may represent a poly expression (with elided types) */
+    public static boolean isPossiblePolyDefault(JCDefaultValue tree, Type clazztype) {
+        JCTypeApply applyTree = TreeInfo.getTypeApplication(tree.clazz);
+        if (applyTree != null) {
+            return applyTree.arguments.isEmpty();
+        } else {
```

Comment:
The return statement is not reachable and so the block is not useful. This is because, Foo<>.default is a syntax error. Given a primitive class Foo<T> we will only support Foo.default involving implicit diamond like inference and will not support Foo<>.default which is what this code is handling.

Reply:
(I think I meant to say return applyTree.arguments.isEmpty(); will never be true.)